### PR TITLE
Add additional helpers for working with resources

### PR DESCRIPTION
**Pull request by tas50**
resource_block_name_if_string returns the name of the resource block if it's a string or nil if it's a variable and we can't parse it

match_resource_type? yields the node object if it finds a resource that matches the name passed

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: lib/rubocop/chef/cookbook_helpers.rb [hunk 1]
```diff
@@ -48,6 +59,27 @@ def method_arg_ast_to_string(ast)
 
       private
 
+      #
+      # given a node object does it lookk like a chef resource or not?
```
Comment (afiune) [inline lib/rubocop/chef/cookbook_helpers.rb]: nit: `s/lookk/look`

File: lib/rubocop/chef/cookbook_helpers.rb [hunk 2]
```diff
@@ -2,24 +2,35 @@ module RuboCop
   module Chef
     # Common node helpers used for matching against Chef Infra Cookbooks
     module CookbookHelpers
+      def resource_block_name_if_string(node)
```
[the remark this answers is not in the document]
Comment (tas50) [reply lib/rubocop/chef/cookbook_helpers.rb]: You'd use in a cop that wanted to analyze the name of a resource. For instance we might want to analyze a file resource to see where we're writing to. We're out of luck if it's a variable, but we something is better than nothing. If it's not a string you get a nil back.

File: lib/rubocop/chef/cookbook_helpers.rb (hunk 1 above)
Comment (tas50) [reply lib/rubocop/chef/cookbook_helpers.rb]: Fixed

File: lib/rubocop/chef/cookbook_helpers.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (lamont-granquist) [reply lib/rubocop/chef/cookbook_helpers.rb]: yeah limitations of static analysis.  if its a string, you're good, if its a variable, nobody knows...

File: lib/rubocop/chef/cookbook_helpers.rb [hunk 3]
```diff
@@ -48,6 +59,27 @@ def method_arg_ast_to_string(ast)
...
 
+      #
+      # given a node object does it look like a chef resource or not?
+      #
+      # @param [RuboCop::AST::Node] node AST object to test
+      #
+      # @return [boolean]
+      #
+      def looks_like_resource?(node)
+        return false unless node.block_type? # resources are blocks if they have properties
```
Comment (lamont-granquist) [inline lib/rubocop/chef/cookbook_helpers.rb]: but what about `package "foo"`?  or is the point mostly to go through the properties so you don't really care?

**Pull request merged**